python/lsst/faro/base/MatchedCatalogBase.py

```diff
@@ -180,18 +181,22 @@ def run(
     ):
         self.log.info("Running catalog matching")
         radius = geom.Angle(self.radius, geom.arcseconds)
-        srcvis, matched = matchCatalogs(
-            sourceCatalogs, photoCalibs, astromCalibs, dataIds, radius, logger=self.log
-        )
-        # Trim the output to the patch bounding box
-        out_matched = type(matched)(matched.schema)
-        self.log.info("%s sources in matched catalog.", len(matched))
-        for record in matched:
-            if box.contains(wcs.skyToPixel(record.getCoord())):
-                out_matched.append(record)
-        self.log.info(
-            "%s sources when trimmed to %s boundaries.", len(out_matched), self.level
-        )
+        if len(sourceCatalogs) < 2:
+            self.log.warning("%s valid input catalogs: ", len(sourceCatalogs))
+            out_matched = afwTable.SimpleCatalog()
```

Comment:
It looks like this is effectively an empty catalog placeholder to handle situations when there is only a single catalog or none at all.